providers/fips/self_test_kats.c

```diff
@@ -193,6 +198,108 @@ static int self_test_kdf(const ST_KAT_KDF *t, OSSL_ST_EVENT *event,
     return ret;
 }
 
+static size_t drbg_kat_entropy_cb(RAND_DRBG *drbg, unsigned char **pout,
+                                  int entropy, size_t min_len, size_t max_len,
+                                  int prediction_resistance)
+{
+    OSSL_PARAM *drbg_params = RAND_DRBG_get_callback_data(drbg);
+    OSSL_PARAM *p = OSSL_PARAM_locate(drbg_params, DRBG_PARAM_ENTROPY);
+
+    if (p == NULL || p->data_type != OSSL_PARAM_OCTET_STRING)
+        return 0;
+    *pout = (unsigned char *)p->data;
+    return p->data_size;
+}
+
+static size_t drbg_kat_nonce_cb(RAND_DRBG *drbg, unsigned char **pout,
+                                int entropy, size_t min_len, size_t max_len)
+{
+    OSSL_PARAM *drbg_params = RAND_DRBG_get_callback_data(drbg);
+    OSSL_PARAM *p = OSSL_PARAM_locate(drbg_params, DRBG_PARAM_NONCE);
+
+    if (p == NULL || p->data_type != OSSL_PARAM_OCTET_STRING)
+        return 0;
+    *pout = (unsigned char *)p->data;
+    return p->data_size;
+}
+
+static int self_test_drbg(const ST_KAT_DRBG *t, OSSL_ST_EVENT *event,
+                          OPENSSL_CTX *libctx)
+{
+    int ret = 0;
+    unsigned char out[256];
+    RAND_DRBG *drbg = NULL;
+    unsigned int flags = 0;
+    int prediction_resistance = 1; /* Causes a reseed */
+    OSSL_PARAM drbg_params[3] = {
+        OSSL_PARAM_END, OSSL_PARAM_END, OSSL_PARAM_END
+    };
+    static const unsigned char zero[sizeof(drbg->data)] = { 0 };
```

Comment:
This looks like a NULL access of drbg?

Reply:
It isn't.

Reply:
As Pauli already noted: it happens at compile time. So unless the compiler is buggy...